src/components/SnapshotDetails/tabs/SnapshotPipelineRunsList.tsx

```diff
@@ -0,0 +1,214 @@
+      >
+        Pipeline runs
+      </Title>
+      <Toolbar data-test="pipelinerun-list-toolbar" clearAllFilters={onClearFilters}>
+        <ToolbarContent>
+          <ToolbarGroup align={{ default: 'alignLeft' }}>
+            <ToolbarItem className="pf-v5-u-ml-0">
+              <SearchInput
+                name="nameInput"
+                data-test="plr-name-filter"
+                type="search"
+                aria-label="name filter"
+                placeholder="Filter by name..."
+                onChange={(e, n) => onNameInput(n)}
+                value={nameFilter}
+              />
+            </ToolbarItem>
+            <ToolbarItem>
+              <Select
+                placeholderText="Type"
+                data-test="plr-type-filter"
+                toggleIcon={<FilterIcon />}
+                toggleAriaLabel="Type filter menu"
+                variant={SelectVariant.checkbox}
+                isOpen={typeFilterExpanded}
+                onToggle={(ev, expanded) => setTypeFilterExpanded(expanded)}
+                onSelect={(event, selection) => {
+                  const checked = (event.target as HTMLInputElement).checked;
+                  setTypeFilters(
+                    checked
+                      ? [...typeFilters, String(selection)]
+                      : typeFilters.filter((value) => value !== selection),
+                  );
+                }}
+                selections={typeFilters}
+                isGrouped
+              >
+                {[
+                  <SelectGroup label="Type" key="type">
+                    {Object.keys(statusFilterObj).map((filter) => (
+                      <SelectOption
+                        key={filter}
+                        value={filter}
+                        isChecked={typeFilters.includes(filter)}
+                        itemCount={statusFilterObj[filter] ?? 0}
+                      >
+                        {filter}
+                      </SelectOption>
+                    ))}
+                  </SelectGroup>,
+                ]}
+              </Select>
+            </ToolbarItem>
+          </ToolbarGroup>
+        </ToolbarContent>
+      </Toolbar>
+      {filteredPLRs.length > 0 ? (
+        <Table
+          data={filteredPLRs}
+          aria-label="Pipeline run List"
+          customData={vulnerabilities}
+          Header={PipelineRunListHeaderWithVulnerabilities}
+          Row={PipelineRunListRowWithVulnerabilities}
+          loaded
+          getRowProps={(obj: PipelineRunKind) => ({
+            id: obj.metadata.name,
+          })}
+          onRowsRendered={({ stopIndex }) => {
+            if (stopIndex === filteredPLRs.length - 1 && !nameFilter) {
+              if (vulnerabilities.fetchedPipelineRuns.length === filteredPLRs.length) {
+                getNextPage?.();
+              } else {
+                if (requestQueue.current.length === 0) {
+                  getNextPage && requestQueue.current.push(getNextPage);
```

Comment:
We add `getNextPage` to the `requestQueue` ref, but it is never de-queued.

Reply:
updated